### PR TITLE
feat: subscription event handling

### PR DESCRIPTION
**Pull request by nduchak**

### CODE REVIEW

File: src/services/notifier/handlers/provider.ts [hunk 1]
```diff
@@ -34,6 +36,34 @@ export const handlers = {
...
+    const providerIns = await ProviderModel.findOne({ where: { provider } })
+
+    if (!providerIns) throw new Error(`Provider ${provider} not found`)
+
+    const [host, port] = providerIns.url.split(/(?::)(\d*)$/, 2)
+    const notifierService = new NotifierSvcProvider({ host, port })
+
+    const subscriptionResponse = await notifierService.getSubscriptions(consumer, [hash])
+
+    const [subscription] = subscriptionResponse
```
Comment (jurajpiar) [inline src/services/notifier/handlers/provider.ts]: is there a reason for this to be in two separate lines?

File: src/services/notifier/hooks/subscriptions.hook.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { HookContext } from '@feathersjs/feathers'
+import { disallow } from 'feathers-hooks-common'
+import { lowerCaseAddressesQueryParamsHook } from '../../utils'
+
+// const ADDRESSES_FIELDS = ['token', 'account']
```
Comment (jurajpiar) [inline src/services/notifier/hooks/subscriptions.hook.ts]: can this be removed?

File: src/services/notifier/hooks/subscriptions.hook.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+export default {
+  before: {
+    all: [
+      (context: HookContext) => {
+        context.params.sequelize = {
+          raw: false
+        }
+      }
+      // lowerCaseAddressesQueryParamsHook(ADDRESSES_FIELDS)
```
Comment (jurajpiar) [inline src/services/notifier/hooks/subscriptions.hook.ts]: can this be removed?

File: src/services/notifier/handlers/provider.ts [hunk 4]
```diff
@@ -34,6 +36,34 @@ export const handlers = {
...
+      consumer,
+      subscriptionId: subscription.id,
+      status: subscription.status,
+      subscriptionPlanId: subscription.subscriptionPlanId,
+      previousSubscription: subscription.previousSubscription,
+      expirationDate: new Date(subscription.expirationDate),
+      topics: subscription.topics
+    })
+
+    if (subscriptionService.emit) subscriptionService.emit('created', wrapEvent('SubscriptionCreated', { provider, consumer, hash }))
```
Comment (jurajpiar) [inline src/services/notifier/handlers/provider.ts]: Shouldn't the emitted object contain all of the data? That would be quite convenient for the client I'd say.

**Review by jurajpiar — CHANGES_REQUESTED** [review]
The change request is solely for the we event emission comment.

File: src/services/notifier/handlers/provider.ts (hunk 1 above)
Comment (nduchak) [reply src/services/notifier/handlers/provider.ts]: nope

**Review by jurajpiar — APPROVED** [review]

**Pull request merged**